### PR TITLE
[iOS] Handle focus changes for UIButton

### LINKED ISSUE #6482: Button Focus and Unfocus can't fire when run on MacOS
### Description

I created a MAUI button and tried to set Focus and Unfous events, but the events won't be fired on MasOS.
The button look like won't get focus if we use tab to shift focus, Click button also does not work.

### Steps to Reproduce

1. Create a new MAUI Project on MacOS,
2. Add a button with Focus and Unfocus events, and set event handler.
3. Run a application and shift focus with 'Tab'

### Version with bug

Release Candidate 1 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

macOS

### Affected platform versions

MacOS 12.0.1

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

Handle button focus changes for UIButton 

code to test on maccatalyst, make sure all events fire when for example using tab key

```
<StackLayout>
        <Label Text="Does this can be focused?"/>
        <Entry Text="Focus Entry" Focused="EntryFocused" Unfocused="EntryUnfocused" />
        <Button Text="Focus Button" Focused="BtnFocused" Unfocused="BtnUnfocused"  />
        <Label Text="Seems to work" />
</StackLayout>
```
### Issues Fixed

Fixes #6482 

### Depends on
- [x] https://github.com/dotnet/maui/issues/16330

### CODE REVIEW

File: src/Core/src/Platform/iOS/MauiButton.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using UIKit;
+
+namespace Microsoft.Maui.Platform
+{
+	internal class MauiButton : UIButton
```
Comment (PureWeen) [inline src/Core/src/Platform/iOS/MauiButton.cs:6]: Can we make this public? It'll be frustrating for users if they can't reuse the control we're using.

Comment (rmarinho) [reply src/Core/src/Platform/iOS/MauiButton.cs:6]: Should i change the handler to MauiButton too ?

Comment (PureWeen) [reply src/Core/src/Platform/iOS/MauiButton.cs]: We can't because AFAIK that'd be too breaking of a change. 

pinging @mattleibow just to sanity check.

**Comment by rmarinho** [comment]
Tried to add Device tests but it didn't work ..

```
[Fact(DisplayName = "Focused event Fires")]
public async Task FocusFires()
{
            var button = new ButtonStub();

            var nativeButtonFocused = await GetValueAsync((IView)button,
            handler =>
            {    
                var mauiBtn = handler.PlatformView as Microsoft.Maui.Platform.MauiButton;
                mauiBtn.Focus(new FocusRequest(true));
                mauiBtn.BecomeFirstResponder();
                return handler.PlatformView.Focused;
            });

            Assert.True(nativeButtonFocused);
            Assert.True(button.IsFocused);
}
```

File: src/Core/src/Platform/iOS/MauiButton.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using UIKit;
+
+namespace Microsoft.Maui.Platform
+{
+	public class MauiButton : UIButton
```
Comment (PureWeen) [inline src/Core/src/Platform/iOS/MauiButton.cs:6]: Unfortunately, it looks like this breaks Button and makes them all get set to custom

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Comment by mattleibow** [comment]
There is no buttondelegate or controlfocusdelegate?

**Comment by rmarinho** [comment]
@PureWeen  we are stuck on this failing test because there's no way to remove the observer when setting the handler just to null.

**Comment by rmarinho** [comment]
So i guess this is blocked in Disconnect not being called, and also the MemoryTests don't take in account a way to remove the subscription . 

Ideas @PureWeen  ?

**Pull request closed without merging**